chat/public/js/src/chat.js

```diff
@@ -361,9 +361,9 @@ function ChatXBlock(runtime, element, init_data) {
      * pause: delays execution with a timeout
      */
     var pause = function(timeout) {
-        return new Promise(function(resolve, reject) {
-            setTimeout(resolve, timeout);
-        });
+        var promise = $.Deferred();
+        setTimeout(promise.resolve, timeout);
+        return promise
```

Comment:
It is done code-wise, I will fix the PR description and I will fix the semicolon also